### PR TITLE
♻️ [2/3] dynamic scheduler structure refactor round 

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

Refactor round `2/3` **code shuffling** no functionality changes:
- creating a proper public interface for `dynamic-scheduler` module (no modules are allowed to import internals)
- outsources some code out of `_events.py` had too many lines of code.
- modules rename and reordering, the `_core` submodule contains most of the internals of the scheduler and its current logic

## Related issue/s

- requires https://github.com/ITISFoundation/osparc-simcore/pull/3698
- https://github.com/ITISFoundation/osparc-issues/issues/638

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_abc.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
+# pylint: disable=relative-beyond-top-level
```
Comment (sanderegg) [inline services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_abc.py:1]: is pylint getting it wrong??

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_observer.py [hunk 2]
```diff
@@ -1,3 +1,5 @@
+# pylint: disable=relative-beyond-top-level
```
Comment (sanderegg) [inline services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_observer.py:1]: are you sure that the pylint configuration is set correctly?

**Review by sanderegg — APPROVED** [review]

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_abc.py (hunk 1 above)
Comment (GitHK) [reply services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_abc.py:1]: Might be a bug with pylint, from what I can tell.
It also sounds very wired to me!

https://github.com/flycheck/flycheck/issues/1758

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_observer.py (hunk 2 above)
Comment (GitHK) [reply services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_observer.py:1]: let's talk on the above discussion

File: services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_abc.py (hunk 1 above)
Comment (GitHK) [reply services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/_core/_abc.py:1]: From pylint' source code, this is a bit misleading.

```
:relative-beyond-top-level (E0402): *Attempted relative import beyond top-level package*
  Used when a relative import tries to access too many levels in the current
  package.
```
[Find above here](https://github.dev/PyCQA/pylint/blob/edc79e5392b0abfed222631501905ecd7521304e/doc/user_guide/checkers/features.rst#L520-L522)

**Pull request merged**